src/Microsoft.OpenApi.Readers/ParseNodes/RootNode.cs

```diff
@@ -18,6 +18,7 @@ internal class RootNode : ParseNode
             YamlDocument yamlDocument) : base(context, diagnostic)
         {
             _yamlDocument = yamlDocument;
+            context.RootNode = this;
```

Comment:
>context.RootNode = this; [](start = 12, length = 24)

Can this assignment be done in ParsingContext?

It's unconventional to the user who calls the constructor `new RootNode(this, diagnostic, yamlDocument)` to expect the state of `this` to be changed.

Reply:
For the default case I believe that line is actually redundant.  See https://github.com/Microsoft/OpenAPI.NET/pull/161/files/5078a58da2d8a942406cb5509192fa9e92569aed#diff-cac6c7a9e1481a7891613d3a4ae1867dR36  It might require changing some tests to ensure the context.RootNode is assigned.

Reply:
Looks like that's the case.

---
In reply to: [157909753](https://github.com/Microsoft/OpenAPI.NET/pull/161#discussion_r157909753) [](ancestors = 157909753)

Reply:
Fixed.